app/scripts/services/secrets.js

```diff
@@ -28,17 +28,29 @@ angular.module("openshiftConsole")
       return secretsByType;
     };
 
+    var setDataParams = function(decodedSecretData, serverName, data) {
+      var usernamePassword = [];
+      if (data.auth) {
+        usernamePassword = window.atob(data.auth).split(":");
+      } else if (data.username && data.password) {
+        usernamePassword = [data.username, data.password];
+      }
+
+      decodedSecretData.auths[serverName] = {
+        username: usernamePassword[0],
+        password: usernamePassword[1],
+        email: data.email
+      };
```

Comment:
I would let the caller set the value in the auths object and just let this function return the data. Then the function takes fewer arguments and doesn't need to know anything about the structure of the `decodedSecretData` object passed to it.

(untested)

```js
var getDockercfgServerParams = function(serverData) {
  var params = _.pick(serverData, ['email', 'username', 'password']);
  if (serverData.auth) {
    try {
      // Decode Base64-encoded username:password string.
      _.spread(window.atob(data.auth).split(":"), function(username, password) {
        params.username = username;
        params.password = password;
      });
    } catch(e) {}
  }

  return params;
};
```

then below in the `decodeDockercfg` function

```js
decodedSecretData.auths[serverName] = getDockercfgServerParams(data);
```

https://lodash.com/docs/4.17.4#pick
https://lodash.com/docs/4.17.4#spread